### PR TITLE
Merge evolved artificial neural network strategy

### PR DESCRIPTION
**Pull request by mojones**

### CODE REVIEW

File: axelrod/strategies/ann.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            total_self_d,
+            turns_remaining
+        ])
+        if output > 0:
+            return C
+        else:
+            return D
+
+
+class EvolvedANN(ANN):
```
Comment (drvinceknight) [inline axelrod/strategies/ann.py]: Can we have more of a docstring here please. 

- Saying that it's based on a neural net etc.
- Include:

    ```
    Names:

         - Evolved ANN: : Original name by Marc Harper/Marting Jones
    ```

File: axelrod/strategies/ann.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    name = "EvolvedANN"
+
+    @init_args
+    def __init__(self):
+        self.classifier['makes_use_of'] = set(['length'])
+        input_values = 17
+        hidden_layer_size = 10
+
+        test_weights = [0.19789658035994948, -5575.476236516673, 0.1028948855131803, 0.7421752484224489,
```
Comment (drvinceknight) [inline axelrod/strategies/ann.py]: Is `test_weights` the right name for this?

File: axelrod/tests/unit/test_ann.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    def test_strategy(self):
+        # Test initial play sequence
+        self.first_play_test(C)
+
+
+class TestEvolvedANNvsCooperator(TestHeadsUp):
+    def test_rounds(self):
+        self.versus_test(axelrod.EvolvedANN(), axelrod.Cooperator(),
+                         [C, D, D, C, D], [C] * 5)
```
Comment (drvinceknight) [inline axelrod/tests/unit/test_ann.py]: Could we throw in a couple more? (Just to ensure we have some more variety).

File: axelrod/strategies/ann.py (hunk 1 above)
Comment (drvinceknight) [inline axelrod/strategies/ann.py]: Perhaps also include a reference (in `bibliography.rst`) to a good base text on neural nets? (Perhaps.)

File: axelrod/strategies/ann.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            my_previous_d = 0 if self.history[-1] == D else 0
+            my_previous2_c = 1 if self.history[-2] == C else 0
+            my_previous2_d = 1 if self.history[-2] == D else 0
+            opponent_previous_c = 1 if opponent.history[-1] == C else 0
+            opponent_previous_d = 1 if opponent.history[-1] == D else 0
+            opponent_previous2_c = 1 if opponent.history[-2] == C else 0
+            opponent_previous2_d = 1 if opponent.history[-2] == D else 0
+
+        # Remaining Features
+        turns_remaining = self.match_attributes['length'] - len(self.history)
```
Comment (drvinceknight) [inline axelrod/strategies/ann.py]: Can we add a test for the case of a match when the match length is unknown? (Just to make sure this doesn't fall over).

Comment (marcharper) [inline axelrod/strategies/ann.py]: The heads up tests don't supply the match length.

Comment (marcharper) [inline axelrod/strategies/ann.py]: We should train a separate strategy that only uses the round number though.

File: axelrod/tests/unit/test_ann.py (hunk 3 above)
Comment (marcharper) [inline axelrod/tests/unit/test_ann.py]: sure

File: axelrod/strategies/ann.py (hunk 2 above)
Comment (marcharper) [inline axelrod/strategies/ann.py]: I changed it to weights.

File: axelrod/strategies/ann.py (hunk 1 above)
Comment (drvinceknight) [inline axelrod/strategies/ann.py]: Sure.

**Pull request merged**